circle/docker-development-image.sh

```diff
@@ -55,7 +46,35 @@ if [[ -n $DOCKER_PASS ]]; then
           docker_build_and_push "${DOCKER_PROJECT}/${IMAGE_NAME}:${push_tag}" "${rs_dir}" "${DOCKER_PROJECT}/${IMAGE_NAME}:${cache_tag}" || exit 1
```

Comment:
Now that we have separated the base images case, the must_exist logic etc. can be removed, and just call `docker_build_and_push`, as that logic is meant for variants